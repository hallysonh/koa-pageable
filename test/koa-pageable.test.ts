import { Context } from 'koa';
import { Order, Direction } from '@hallysonh/pageable';
import paginate from '../src/koa-pageable';

describe('Tests', () => {
  describe('paginate function', () => {
    const next = () => ({});
    const context = {
      query: {
        page: 1,
        size: 20,
        sort: 'firstName',
        indexed: 'true',
      },
      state: {},
    } as Context;

    it('context.state matches snapshot when paginate is called with valid context', async () => {
      await paginate(context, next);
      expect(context.state).toMatchSnapshot();
    });

    it('context.state matches snapshot when paginate is called with string values for page and size', async () => {
      const updatedQuery = { ...context.query, page: '10', size: '15' };
      await paginate({ ...context, query: updatedQuery }, next);
      expect(context.state).toMatchSnapshot();
    });

    it('context.state matches snapshot when paginate is called with empty strings for page and size', async () => {
      const updatedQuery = { ...context.query, page: '', size: '' };
      await paginate({ ...context, query: updatedQuery }, next);
      expect(context.state).toMatchSnapshot();
    });

    it('context.state matches snapshot when paginate is called with undefined page, size, and sort', async () => {
      const updatedQuery = { indexed: 'true' };
      await paginate({ ...context, query: updatedQuery }, next);
      expect(context.state).toMatchSnapshot();
    });

    it('throws error when paginate is called with invalid values for page and size', async () => {
      const updatedQuery = { ...context.query, page: 'page', size: 'size' };
      try {
        await paginate({ ...context, query: updatedQuery }, next);
      } catch (e) {
        expect(e.message).toBe("Could not convert 'page' to number");
      }
    });
  });
});
