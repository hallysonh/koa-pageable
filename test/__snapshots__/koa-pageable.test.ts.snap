// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests paginate function context.state matches snapshot when paginate is called with empty strings for page and size 1`] = `
Object {
  "pageable": Object {
    "indexed": true,
    "page": 0,
    "size": 10,
    "sort": Array [
      e {
        "direction": "asc",
        "property": "firstName",
      },
    ],
  },
}
`;

exports[`Tests paginate function context.state matches snapshot when paginate is called with string values for page and size 1`] = `
Object {
  "pageable": Object {
    "indexed": true,
    "page": 10,
    "size": 15,
    "sort": Array [
      e {
        "direction": "asc",
        "property": "firstName",
      },
    ],
  },
}
`;

exports[`Tests paginate function context.state matches snapshot when paginate is called with undefined page, size, and sort 1`] = `
Object {
  "pageable": Object {
    "indexed": true,
    "page": 0,
    "size": 10,
  },
}
`;

exports[`Tests paginate function context.state matches snapshot when paginate is called with valid context 1`] = `
Object {
  "pageable": Object {
    "indexed": true,
    "page": 1,
    "size": 20,
    "sort": Array [
      e {
        "direction": "asc",
        "property": "firstName",
      },
    ],
  },
}
`;
